### DISCUSSION TITLE
Average cost per month (EUR)

### DISCUSSION THREAD

**Discussion by dbrennand** [vitobotta/hetzner-k3s#130] [General]
Hi there,

This looks like a great tool, I've taken a look at the README, issues and Wiki but I can't find any mention on how much it costs per month to run a basic HA cluster?

Would someone be able to provide some numbers?

I know it depends on various factors but just an idea of how much would be great to know 🙂

Thanks!

**Comment by vitobotta** [comment]
Hi! Well yes it depends on 

- whether you are ok with a single master or want a HA control plane with multiple masters
- a HA control plane requires a load balancer for the API server, so it's an additional cost compared to running a single master
- how much CPU and memory your workloads require

So you Need to ask yourself these questions to get an idea of the cost. As an example, a cluster with 
- a single master CX21 (2 cores, 4 GB of RAM)
- 3 worker nodes CPX31 (4 cores, 8GB of RAM)
- a load balancer for the ingress controller
- around 100-120GB of storage

cost me around 50-55 euros per month if I remember correctly. You may need more or less depending on your workloads. What are you planning on running on the cluster?

Check the Hetzner pricing page at https://www.hetzner.com/cloud

**Reply by dbrennand** [reply]
Hi @vitobotta 

Thanks for the quick reply!

I see, OK well with these questions in mind probably a single node control plane and a single worker to start with as the workload wouldn't be critical or anything like that. Mostly experimental stuff 🙂

Also, that example is exactly what I was looking for so thanks for that!

**Reply by vitobotta** [reply]
You can also specify only node pools with autoscaling enabled, so you only use the nodes you actually need at all times instead of worrying about how many nodes you need :)
